backend/app/adapter/google/identityprovider.go

```diff
@@ -0,0 +1,75 @@
+package google
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"short/app/usecase/service"
+
+	"github.com/byliuyang/app/fw"
+)
+
+const (
+	authorizationAPI     = "https://accounts.google.com/o/oauth2/v2/auth?"
```

Comment:
```suggestion
	authorizationAPI     = "https://accounts.google.com/o/oauth2/v2/auth"
```

Reply:
The `?` is used to separate query parameters from the URL, which shouldn't be included here.